### PR TITLE
Xcode project cleanup round 2

### LINKED ISSUE #57: Clean up Xcode project
Something up, both schemes seem to list macOS, iOS, and watchOS targets for running/testing.  So something setting was is confusing Xcode.  It would be nice to figure that out and get a project that acts a little more normal.

### PR DESCRIPTION
**Pull request by thomasvl**
Fixes #57 

Everything is done this time around as small sub commits incase there are issues after the fact.  Also makes it more clear what was done and why.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -235,3 +250,40 @@ regenerate-test-protos: ${PROTOC_GEN_SWIFTX}
...
+	  -scheme SwiftProtobufRuntime_iOS \
+	  -configuration Release \
+	  -destination "platform=iOS Simulator,name=iPhone 6s,OS=latest" \
+	  test
+
+test-xcode-macOS-debug:
+	xcodebuild -project SwiftProtobufRuntime.xcodeproj \
+	  -scheme SwiftProtobufRuntime_macOS \
+	  -configuration debug \
+	  build test
```
Comment (tbkka) [inline Makefile]: Is there a reason the macOS test targets use "build test" and the iOS ones use only "test" here?

File: SwiftProtobufRuntime.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -961,309 +959,190 @@
...
-				ONLY_ACTIVE_ARCH = YES;
-				PRODUCT_BUNDLE_IDENTIFIER = "Apple.ProtobufTestSuite-iOS";
+				LD_RUNPATH_SEARCH_PATHS = "@loader_path/Frameworks";
+				PRODUCT_BUNDLE_IDENTIFIER = com.apple.protobuf.SwiftProtobufRuntimeTests;
 				PRODUCT_NAME = "$(TARGET_NAME)";
 				SDKROOT = iphoneos;
-				SWIFT_ACTIVE_COMPILATION_CONDITIONS = DEBUG;
-				SWIFT_OPTIMIZATION_LEVEL = "-Onone";
-				SWIFT_VERSION = 3.0;
+				TARGETED_DEVICE_FAMILY = "1,2";
```
Comment (tbkka) [inline SwiftProtobufRuntime.xcodeproj/project.pbxproj]: One concern:   Testing only iOS and macOS won't cover any 32-bit platforms.  Maybe we should add watchOS tests?

File: Makefile (hunk 1 above)
Comment (thomasvl) [inline Makefile]: Cause Xcode.  `xcodebuild` will honor the `build` action a macOS framework, but it won't for an iOS framework.  I tend to try to use both just incase we add other things to the actions in the future.

File: SwiftProtobufRuntime.xcodeproj/project.pbxproj (hunk 2 above)
Comment (thomasvl) [inline SwiftProtobufRuntime.xcodeproj/project.pbxproj]: I've got another branch queued up to add 32bit iOS destinations to the tests also.  The initial Xcode project didn't really support it, so I wanted to land these cleanups before changing things in the project.  It's on my other machine, so hopefully I'll get a chance to send out the PR later today.

**Pull request merged**